gradio/components/image.py

```diff
@@ -25,6 +26,28 @@
 PIL.Image.init()  # fixes https://github.com/gradio-app/gradio/issues/2843
 
 
+class ImageData(GradioModel):
```

Comment:
I'm confused why we still need the `ImageData` now that we have the `Base64ImageData` class. Either the data is being sent as a file, in which case the path parameter is present and we can use `FileData`, or its being sent as base64, in which case, we can use the `Base64ImageData` dataclass

If anything, ImageData should just be a union of `FileData` and `Base64ImageData`

Reply:
FileData requires `path` to be a non-empty string. That's not the case when we are streaming base64 to the server. So if we use FileData and send base64 without a path key, the server will raise an error. Base64ImageData was added to provide more specific api info for output image components.